src/main/java/uk/gov/hmcts/darts/cases/helper/AdvancedSearchRequestHelper.java

```diff
@@ -71,12 +71,12 @@ private List<Predicate> createPredicates(GetCasesSearchRequest request, Criteria
         List<Predicate> predicates = new ArrayList<>();
         CollectionUtils.addAll(predicates, createCaseCriteria(request, criteriaBuilder, caseRoot));
         CollectionUtils.addAll(predicates, addHearingDateCriteria(request, criteriaBuilder, caseRoot));
-        CollectionUtils.addAll(predicates, addCourthouseCriteria(request, criteriaBuilder, caseRoot));
         CollectionUtils.addAll(predicates, addCourtroomCriteria(request, criteriaBuilder, caseRoot));
         CollectionUtils.addAll(predicates, addJudgeCriteria(request, criteriaBuilder, caseRoot));
         CollectionUtils.addAll(predicates, addDefendantCriteria(request, criteriaBuilder, caseRoot));
         CollectionUtils.addAll(predicates, addEventCriteria(request, criteriaBuilder, caseRoot));
         if (!userIdentity.userHasGlobalAccess(Set.of(JUDGE))) {
+            CollectionUtils.addAll(predicates, addCourthouseCriteria(request, criteriaBuilder, caseRoot));
             CollectionUtils.addAll(predicates, addUserSecurityRolesCriteria(criteriaBuilder, caseRoot));
```

Comment:
The `addUserSecurityRolesCriteria` method maybe needs to be renamed as it doesn't seem to be looking at security roles, but security groups to determine the user's courthouses via the `joinUser` method?

Reply:
renamed - I don't know if the original was joined by roles - but changed to groups now